src/utils/plotSpecs/generateViolinSpec.js

```diff
@@ -0,0 +1,475 @@
+                  field: 'y',
+                },
+                size: {
+                  value: config.marker.size,
+                },
+                strokeWidth: {
+                  value: 0,
+                },
+                opacity: {
+                  value: 1,
+                },
+                fill: {
+                  value: 'black',
+                },
+                fillOpacity: {
+                  value: config.marker.opacity / 10,
+                },
+              },
+            },
+          },
+        ],
+      },
+    ],
+    title:
+    {
+      text: config.title.text ? config.title.text : config.shownGene,
+      color: { value: config.colour.masterColour },
+      anchor: { value: config.title.anchor },
+      font: { value: config.fontStyle.font },
+      dx: { value: config.title.dx },
+      fontSize: { value: config.title.fontSize },
+    },
+  };
+  // So far, signals cannot be used to hide stuff
+  //   https://github.com/vega/vega/issues/1153
+  // so we remove the stuff to be hidden by changing the spec
+
+  if (!config.selectedPointsVisible) {
+    spec.marks[0].marks = spec.marks[0].marks.filter((mark) => mark.from.data !== 'cellsInGroup');
+  }
+  if (!config.statisticsVisible) {
+    spec.marks[0].marks = spec.marks[0].marks.filter((mark) => mark.from.data !== 'summaryOfGroup');
+  }
+
+  if (config?.legend.enabled) {
+    const positionIsRight = config.legend.position === 'right';
+
+    const legendColumns = positionIsRight ? 1 : Math.floor(config.dimensions.width / 85);
+    const labelLimit = positionIsRight ? 0 : 85;
+    if (positionIsRight) {
+      const plotWidthIndex = spec.signals.findIndex((item) => item.name === 'plotWidth');
+      spec.signals[plotWidthIndex].value = plotWidth * 0.85;
+    }
+
+    const groups = _.keys(plotData.groups);
+    const groupNames = groups.map((id) => plotData.groups[id].name);
+    const groupColors = groups.map((id) => plotData.groups[id].color);
+    spec.scales.push({
+      name: 'legend',
+      type: 'ordinal',
+      range: groupColors,
+      domain: groupNames,
+    });
+    spec.legends = [
+      {
+        fill: 'legend',
+        type: 'symbol',
+        symbolType: 'square',
+        symbolSize: { value: 200 },
+        orient: config?.legend.position,
+        offset: 40,
+        direction: 'horizontal',
+        labelFont: { value: config?.fontStyle.font },
+        columns: legendColumns,
+        labelLimit,
+      },
+    ];
+  }
+
+  return spec;
+};
+
+const generateData = (
+  cellSets,
+  selectedExpression,
+  groupingHierarchyId,
+  displayId,
+) => {
+  /*
+   groups: {
+      group_id_1: {
+        name: cellSet.properties[group_id_1].name,
+        color: cellSet.properties[group_id_1].color,
+      },
+      ...
+    },
+    cells: [
+      {
+        group: group_id_1,
+        y: selectedExpression[cellId],
+        x: Math.random() // only if cell has to be displayed
+      },
+      ...
+    ]
+  */
```

Comment:
no need for that

Reply:
It is not dead code, but rather a comment documenting the format. I updated it to make it clearer.